multiqc/modules/base_module.py

```diff
@@ -281,14 +281,70 @@ def add_section(
+        if len(set(clean_names)) == 1:  # all the same
+            return clean_names[0]
+
+        if len(clean_names) == 2:
+            # Possibly a FASTQ pair - try trimming FASTQ suffixes.
+            r1, r2 = sorted(clean_names)
+
+            # Try trimming the conventional illumina suffix with a tail 001 ending. Refs:
+            # https://support.illumina.com/help/BaseSpace_Sequence_Hub_OLH_009008_2/Source/Informatics/BS/NamingConvention_FASTQ-files-swBS.htm
+            # https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/fastq-input#:~:text=10x%20pipelines%20need%20files%20named,individual%20who%20demultiplexed%20your%20flowcell.
+            cleaned_r1 = re.sub(r"_R1_\d\d\d$", "", r1)
+            cleaned_r2 = re.sub(r"_R2_\d\d\d$", "", r2)
+            if cleaned_r1 == cleaned_r2:  # trimmed successfully
+                return cleaned_r1
+
+            # Try removing _R1 and _R2 from the middle.
+            cleaned_r1 = re.sub(r"_R1_", "_", r1)
+            cleaned_r2 = re.sub(r"_R2_", "_", r2)
+            if cleaned_r1 == cleaned_r2:  # trimmed successfully
+                return cleaned_r1
+
+            # Try trimming other variations from the end (-R1, _r1, _1, .1, etc).
+            cleaned_r1 = re.sub(r"([_.-][rR]?1)?$", "", r1)
+            cleaned_r2 = re.sub(r"([_.-][rR]?2)?$", "", r2)
+            if cleaned_r1 == cleaned_r2:  # trimmed successfully
+                return cleaned_r1
+
+        # More than 2 names, so can't assume a FASTQ pair. Just trying to keep a common prefix.
+        prefix = os.path.commonprefix(clean_names)
+        prefix = prefix.rstrip("_.- ")
+        # If the prefix is empty, join the names with a dash.
+        if prefix == "":
+            s_name = "_".join(clean_names)
+        else:
+            s_name = prefix
+        return s_name
+
+    def clean_s_name(self, s_name: Union[str, List[str]], f=None, root=None, filename=None, seach_pattern_key=None):
+        """
+        Helper function to take a long file name(s) and strip back to one clean sample name. Somewhat arbitrary.
+        :param s_name: The sample name(s) to clean.
         :param root: The directory path that this file is within
         :config.prepend_dirs: boolean, whether to prepend dir name to s_name
         :return: The cleaned sample name, ready to be used
         """
+        if isinstance(s_name, list):
+            if len(s_name) == 0:
+                raise ValueError("Empty list of sample names passed to clean_s_name()")
+            if len(s_name) >= 2:
+                # Extract a sample name from a list of file names (for example, FASTQ pairs).
+                # Each name is cleaned separately first.
+                clean_names = [
+                    self.clean_s_name(sn, f=f, root=root, filename=filename, seach_pattern_key=seach_pattern_key)
+                    for sn in s_name
+                ]
+                return self._clean_fastq_name_pair(clean_names)
```

Comment:
I think that we don't want to return yet, no?

eg. if two identical filenames `sample.fastq.gz` and `sample.fastq.gz` are given, it'll return the full `sample.fastq.gz` instead of cleaning to `sample`.

Also, returning at this stage means we miss directory prefixes etc.

Reply:
> eg. if two identical filenames `sample.fastq.gz` and `sample.fastq.gz` are given, it'll return the full `sample.fastq.gz` instead of cleaning to `sample`.

It will be cleaned to two identical [`sample`, `sample`] in the list comprehension above. That's why the variable is called `clean_names` :)

Reply:
Ooh I didn't notice that - calling the function on itself, nice!